031-tls-encrypt-the-kafka-connect-rest-api.md

```diff
@@ -0,0 +1,156 @@
+# TLS encrypting the Kafka Connect REST API
+
+This proposal adds the option to reduce access to the Kafka Connect REST API by enabling TLS encryption on the Kafka Connect REST listener.
+
+## Current situation
+
+Currently, instances of Kafka Connect that are deployed by the Strimzi operators are configured with the default REST API endpoint settings.
+This means that the Kafka Connect REST API endpoint uses HTTP on port 8083 and that the Strimzi `KafkaConnect`, `KafkaConnector` and `KafkaMirrorMaker2` operators make unsecured REST client calls based on this default configuration.
+
+The default network policies created by the Strimzi operators restrict incoming REST API calls to only allow access from the operator pod.
+Users can define further network policies to override the default policy and allow wider access.
+
+## Motivation
+
+There is currently no way to TLS encrypt the Kafka Connect REST API, leaving the Connect REST listener as perhaps the last key endpoint that cannot currently be restricted using Strimzi.
+
+## Proposal
+
+This proposal adds REST api configuration options to the `KafkaMirrorMaker2` and `KafkaConnect` CRDs.
+
+A new field is added to the `KafkaConnect` and `KafkaMirrorMaker2` specs named `restListeners`, which allows users to configure an HTTP listener on port 8443, an encrypted HTTPS listener on port 8443, or both in the Kafka Connect configuration.
```

Comment:
which ports are used when both HTTP and HTTPS are enabled? It cannot be the same 8443 for both, or? Should be HTTP using 8083?

Reply:
Yes, that's a mis-type, HTTP will use 8083 and HTTPS will use 8443. I'll update the proposal.